kubernetes/kubernetes.go

```diff
@@ -261,17 +260,19 @@ func (in *K8SClient) IsExpGatewayAPI() bool {
 			K8sTCPRouteType: "tcproutes",
 			K8sTLSRouteType: "tlsroutes",
 		}
-		isGatewayAPIV1Alpha2 := checkGatewayAPIs(in, K8sNetworkingGroupVersionV1Alpha2.String(), v1alpha2Types)
+		isGatewayAPIV1Alpha2 := checkGatewayAPIs(in, K8sNetworkingGroupVersionV1Alpha2.String(), v1alpha2Types, true)
 		in.isExpGatewayAPI = &isGatewayAPIV1Alpha2
 	}
 	return *in.isExpGatewayAPI
 }
 
-func checkGatewayAPIs(in *K8SClient, version string, types map[string]string) bool {
+func checkGatewayAPIs(in *K8SClient, version string, types map[string]string, isExperimental bool) bool {
 	found := 0
 	res, err := in.k8s.Discovery().ServerResourcesForGroupVersion(version)
 	if err != nil {
-		log.Debugf("Error while checking K8s Gateway API CRDs: %s. Required K8s Gateway API version: %s. Gateway API will not be used.", version, err.Error())
+		if !isExperimental {
+			log.Debugf("K8s Gateway API CRDs are not installed. Required K8s Gateway API version: %s. Gateway API will not be used.", version)
```

Comment:
Are both V1 and V1Beta1 really `Required`?  It seems like unless both exist the user will bet a message saying that  one of them is required and missing?

Reply:
Yes some of objects are not V1 yet, but still required to be. These all objects exists all-together in CRD file.
Only exception is experimental ones, which I made optional and not showing warning anymore.